### PR TITLE
refactor: consolidate pool implementation

### PR DESCRIPTION
**Pull request by alexanderbez**
Closes: #1225

- Move balancer and amm logic into a single `pool.go` file
- Move `PoolParams` to `pool_params.go` file
- Update test package to `balancer_test`
- Make `SetInitialPoolAssets` a public API (to satisfy the `TestBalancerPoolUpdatePoolAssetBalance` test)

### CODE REVIEW

**Comment by alexanderbez** [comment]
@ValarDragon the question remains, do we want `pool_test.go` to be in package `balancer` or `balancer_test`? If the latter, we need to make `Pool#setInitialPoolAssets` a public API.

**Comment by ValarDragon** [comment]
I'm down for it to be in either package tbh

File: x/gamm/pool-models/balancer/pool_params.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package balancer
```
Comment (mattverse) [inline x/gamm/pool-models/balancer/pool_params.go:1]: I love how we made a separate file for pool params!

**Review by mattverse — COMMENTED** [review]
Per deleting `amm.go`, did we decide to remove separating the amm logic and the pool logic by using two different files `amm.go` and `pool.go`?

**Review by mattverse — CHANGES_REQUESTED** [review]
Can we also change stableswap package from stableswap_pool.go -> pool.go to better achieve consolidation between the packages? I'm also down to leaving balancer_pool.go

**Comment by alexanderbez** [comment]
I'm going to need to rebase this again unfortunately, a bit annoying but w.e. The idea is just to consolidate the pool logic into a single file along with the tests.

**Comment by mattverse** [comment]
Oh interesting, I thought we were trying to separate out the pool methods and the core math logic in two different files? 

If we were to change the decision to move towards a single pool.go file, I think we should make the same changes in stableswap package as well or do you think that change is out of scope for this PR?

**Comment by alexanderbez** [comment]
Conflicts are too large here to warrant resolving them. Instead re-opened a new PR: https://github.com/osmosis-labs/osmosis/pull/1868

**Pull request closed without merging**